src/go/k8s/pkg/resources/statefulset_scale.go

```diff
@@ -181,12 +181,12 @@ func (r *StatefulSetResource) handleDecommissionInProgress(ctx context.Context,
 //
 // Before completing the process, it double-checks if the node is still not registered, for handling cases where the node was
 // about to start when the decommissioning process started. If the broker is found, the process is restarted.
-func (r *StatefulSetResource) handleDecommission(ctx context.Context) error {
+func (r *StatefulSetResource) handleDecommission(ctx context.Context, l logr.Logger) error {
 	brokerID := r.pandaCluster.GetDecommissionBrokerID()
 	if brokerID == nil {
 		return nil
 	}
-	log := r.logger.WithName("handleDecommission").WithValues("node_id", *brokerID)
+	log := l.WithName("handleDecommission").WithValues("node_id", *brokerID)
```

Comment:
Why `r.logger` is changed to function parametr? What is the gain here?

Reply:
WithName collections

Reply:
I don't see the benefit. You can always call `r.logger.WithName`.

Reply:
How do you get the list of names that brought you here that way? Afaik, that will always be the base logger.

Reply:
WithName append to the logger? I forgot about it. Resolved.